### DISCUSSION TITLE
Audio events

### DISCUSSION THREAD

**Discussion by WardPearce** [mediagrid/capacitor-native-audio#3] [2. Feature Ideas]
Possibility to be able to add a listener for events like current time update etc

**Comment by wsamoht** [comment]
I've thought about adding an event for current time as it would be a tiny bit faster than calling the method `getCurrentTime` because it would reduce it from a request/response to native land to just a response. For the time being, I just use a `setInterval`. See the [code](https://github.com/mediagrid/capacitor-native-audio/blob/main/example/src/js/example.ts#L113) in the example project.

What other events are you looking for? Buffering is on the list to do :)

**Reply by WardPearce** [reply]
Really I just need a way to implement logic to skip sponsors in audio what requires checking the current time continuously until it reaches a time marked as a sponsor. But really the plugin should expose events what people would expect a audio/player element to have in my opinion.

**Comment by WardPearce** [comment]
Also on a different note, I've noticed if you minimise a app but do so by going to the view what shows all the minimised apps (no idea what this is called), it doesn't trigger audio to play.

**Reply by wsamoht** [reply]
Not exactly sure what you mean. Are you talking on Android or iOS? And what do you mean "doesn't trigger audio to play"?

**Reply by WardPearce** [reply]
On Android sorry, so if I create a audio source to play in the background what auto plays using the out of focus event, this works fine when minimising the app normally. But if I go to the "Recents screen", what also minimises the app the out of focus event Isn't fired

**Reply by wsamoht** [reply]
You are talking about the `onAppLosesFocus` event right? I see from your other [discussion](https://github.com/mediagrid/capacitor-native-audio/discussions/2), you are using this only if in the background and an HTML player when the app is in focus. That is an interesting use case 😄 Why not just use it for both? The original purpose of that event was for stopping unnecessary processing in JS when the app is in the background like getting the current time.

We are using Android's [`onStop`](https://developer.android.com/reference/android/app/Activity#onStop()) event for losing focus.

> This is a good place to stop refreshing UI, running animations and other visual things.

**Reply by WardPearce** [reply]
While in my mind the video element is already downloading the audio for video and handling all the syncing for me. And this way I only needs to download the audio again once its out of focus (unless if thats not how it buffers) & also means I don't need to relay on this plugin for audio on web & desktop

**Reply by wsamoht** [reply]
Buffering should start when `initialize` is called. This plugin doesn't support web yet so it wouldn't work anyways :) You would have to keep your implementation for web and only use this plugin for mobile devices.

You could try using Capacitor's `pause` listener - https://capacitorjs.com/docs/apis/app#addlistenerpause-.

**Reply by WardPearce** [reply]
> Buffering should start when `initialize` is called. This plugin doesn't support web yet so it wouldn't work anyways :) You would have to keep your implementation for web and only use this plugin for mobile devices.
> 
> You could try using Capacitor's `pause` listener - https://capacitorjs.com/docs/apis/app#addlistenerpause-.

okay cool I'll check it out, ty
